### PR TITLE
Minor documentation and cleanup of TOTG plugin

### PR DESCRIPTION
**Pull request by AndyZe**
Pulling some changes out of #571 so they can be merged separately.

- Document some limitations of TOTG in the plugin description.
- Minor code cleanup.
- Mark iterative spline and iterative parabola plugins as deprecated since they have some known issues (https://github.com/ros-planning/moveit/issues/1711).

### CODE REVIEW

File: moveit_ros/planning/planning_request_adapters_plugin_description.xml [hunk 1]
```diff
@@ -33,17 +33,19 @@
 
   <class name="default_planner_request_adapters/AddTimeParameterization" type="default_planner_request_adapters::AddTimeParameterization" base_class_type="planning_request_adapter::PlanningRequestAdapter">
     <description>
+      A deprecated time parameterization plugin.
```
Comment (sjahr) [inline moveit_ros/planning/planning_request_adapters_plugin_description.xml:36]: So does this mean TOTG is the only algorithm recommended to use? Maybe this information should be added to the tutorials as well.

Comment (AndyZe) [reply moveit_ros/planning/planning_request_adapters_plugin_description.xml:36]: Yep, I agree.

Comment (henningkayser) [reply moveit_ros/planning/planning_request_adapters_plugin_description.xml:36]: I don't think anyone will ever notice this description. Could you add some warnings if the old plugins are being used? I'm also not convinced the old plugins should be deprecated, switching the default to TOTG and hinting on the shortcomings seems sufficient to me.

Comment (AndyZe) [reply moveit_ros/planning/planning_request_adapters_plugin_description.xml:36]: In the interest of getting this merged, I'll just revert the changes to `TimeParameterization` and `IterativeSplineParameterization`

File: moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp [hunk 2]
```diff
@@ -49,7 +49,8 @@ namespace trajectory_processing
 static const rclcpp::Logger LOGGER =
     rclcpp::get_logger("moveit_trajectory_processing.time_optimal_trajectory_generation");
 
-constexpr double EPS = 0.000001;
+constexpr double DEFAULT_TIMESTEP = 0.001;
```
Comment (tylerjw) [inline moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp:52]: [nit] use `1e-3` to mach the style of the next line.  Also, put this in an anonymous namespace unless it is an exported symbol somewhere?

**Review by tylerjw — APPROVED** [review]
Generally, approve this PR.  Thank you for separating out this small change!  :rocket:

**Review by tylerjw — APPROVED** [review]

**Pull request merged**